Topics/Development_Process/Build_Requirements/CMakeLists_txt.md

```diff
@@ -0,0 +1,199 @@
+
+## Creating CMakeLists.txt
+
+### CMake Language
+`CMakeLists.txt` is written in [CMake Language](https://cmake.org/cmake/help/latest/manual/cmake-language.7.html#manual:cmake-language(7)), which consists of comments, [commands](https://cmake.org/cmake/help/latest/manual/cmake-commands.7.html#manual:cmake-commands(7)), and [variables](https://cmake.org/cmake/help/latest/manual/cmake-variables.7.html#manual:cmake-variables(7)). 
+
+* **Comments** start with `#`
+  ```
+  # This is a comment
+  ```
+* **Commands** start with command names and a list of whitespace-separated arguments (case-sensitive).
+  ```
+  find_package(LIBIGL REQUIRED QUIET)
+  ```
+* **Variables** are case-sensitive with only alphanumeric characters and underscores. 
+
+  A number of useful variables are automatically defined by CMake (e.g `CMAKE_CURRENT_SOURCE_DIR`, `PROJECT_NAME`). 
+
+  Use *set* command to set variable names, and *${variable_name}* to reference a varible in command arguments. 
+  ```
+  set(Foo a b c)    # Value of Foo is "a;b;c"
+
+  command(${Foo})   # Foo is replaced by a;b;c 
+                    # and expands to 3 arguments
+  ```
+
+It also provides the basic **Flow Control** - Conditional statements, Looping constructs, and Procedure definitions. Here are the examples:
+
+* **Conditional statements (e.g [if](https://cmake.org/cmake/help/latest/command/if.html#command:if))**
+    ```
+    if(<condition>)
+    <commands>
+    elseif(<condition>)     # optional block, can be repeated
+    <commands>
+    else()                  # optional block
+    <commands>
+    endif()
+    ```
+* **Looping Constructs (e.g [foreach](https://cmake.org/cmake/help/latest/command/foreach.html#command:foreach), [while](https://cmake.org/cmake/help/latest/command/while.html#command:while))**
+    ```
+    foreach(<variable>                  # <variable> to be referenced
+            <list of variables>)        # <list of variables> to replace ${variable}
+    command(${variable} <arguments>)    # ${variable} - current value from the list
+    endforeach()
+
+    while(<condition>)
+    <commands>
+    endwhile()
+    ```
+* **Procedure definitions (e.g [function](https://cmake.org/cmake/help/latest/command/function.html#command:function), [macro](https://cmake.org/cmake/help/latest/command/macro.html#command:macro))**
+    ```
+    function(<function_name> <parameters>)
+    <commands>
+    endfunction()
+
+    macro(<macro_name> <parameters>)
+    <commands>
+    endmacro()
+    ```
+
+For more detailed knowledge of CMake Language, check the **highlighted links** above.
+
+### Key Components and Concepts
+
+A `CMakeLists.txt` typically consists of the following elements:
```

Comment:
Might want to move further up? Think it might be good to have a "Hello World" equivalent early on so that the reader has a general idea of what programs look like, but ultimately it's your call.